examples/add_pyramid.py

```diff
@@ -0,0 +1,32 @@
+"""
+Displays an image pyramid
```

Comment:
I think an in-memory pyramid is an unlikely scenario, so if performance is better with zarr, why not create a zarr file with this dataset as the first part of the example?

Reply:
So I've done some testing with zarr, identical data, and using timeit and the difference doesn't seem to be that great, though there were a couple places where it was a bit faster, and the times are right around what is noticeable for a human - i.e. a factor of 2x can be huge in this context, as can a difference of 5-10 ms. It's still unclear to me why the one example I have on my laptop seems less laggy than this one, and unfortunately `cProfile` isn't being very helpful, it's not really catching the layer updating, which `timeit` does catch. Not sure if thats something with how `cProfile` does events / multiple threads. Anyway, for now I'd say we just live with the performance, as it will be hard for me to debug without `cProfile` working properly

Reply:
https://github.com/sumerc/yappi this might be a solution

Reply:
Hmm just gave `yappi` a try too and it's only reporting one thread and still missing the functional calls on the layers that are triggered by the event and that `timeit` is including when it measures the total duration of the calls. I'm at a loss